napari/_qt/qt_main_window.py

```diff
@@ -528,7 +528,10 @@ def add_dock_widget(
             # Keep the dropdown menus in the widget in sync with the layer model
             # if widget has a `reset_choices`, which is true for all magicgui
             # `CategoricalWidget`s
-            self.qt_viewer.viewer.layers.events.connect(widget.reset_choices)
+            layer_events = self.qt_viewer.viewer.layers.events
+            layer_events.inserted.connect(widget.reset_choices)
+            layer_events.removed.connect(widget.reset_choices)
+            layer_events.reordered.connect(widget.reset_choices)
```

Comment:
```suggestion
            layers_events = self.qt_viewer.viewer.layers.events
            layers_events.inserted.connect(widget.reset_choices)
            layers_events.removed.connect(widget.reset_choices)
            layers_events.reordered.connect(widget.reset_choices)
```
Might be a little more clear